cxx/util_io.cc

```diff
@@ -11,36 +12,166 @@
 #include <unordered_set>
 #include <vector>
 
+void verify_noisy_relation_domains(const T_schema& schema) {
+  for (const auto& [relname, trelation] : schema) {
+    std::visit(
+        [&](const auto& trel) {
+          using T = std::decay_t<decltype(trel)>;
+          if constexpr (std::is_same_v<T, T_noisy_relation>) {
+            std::vector<std::string> base_domains = std::visit(
+                [&](const auto& brel) { return brel.domains; },
+                schema.at(trel.base_relation));
+            for (size_t i = 0; i != base_domains.size(); ++i) {
+              assert(base_domains[i] == trel.domains[i] &&
+                     "The first domains of a noisy relation must be the same "
+                     "as the domains of the base relation.");
+            }
+          }
+        }, trelation);
+  }
+}
+
 T_schema load_schema(const std::string& path) {
```

Comment:
Could you add the new test in `util_io_test` from #95 that tests the `with_noisy` schema?

Reply:
Done.